src/binary-reader.cc

```diff
@@ -1649,10 +1652,14 @@ static void read_custom_section(Context* ctx, uint32_t section_size) {
         ++i;
         break;
       default:
-        /* unknown subsection, skip rest of section */
-        ctx->offset = ctx->read_end;
+        /* unknown subsection, skip it */
```

Comment:
Another thing we might want to do is to check that subsection data is actually contained in the subsection (by setting ctx->read_end to the subsection end temporarily, for example); just like read_function_body verifies a function body does not extend past its indicated length.